lib/puppet/provider/package/homebrew.rb

```diff
@@ -8,18 +8,20 @@
   def install
     name = install_name
 
-    Puppet.debug "Installing #{name}"
-    output = execute([command(:brew), :install, name, *install_options])
-
-    if output =~ /Searching taps/
-      Puppet.debug "Falling back to brew-cask (still installing #{name}"
-      output = execute([command(:brew), :cask, :install, name, *install_options])
-      # brewcask includes some funky beer characters that f*ck with encoding
-      output = output.encode('UTF-8', :invalid => :replace, :undef => :replace)
-
+    begin
+      Puppet.debug "Looking for #{name} package on brew..."
+      output = execute([command(:brew), :info, name])
       if output.empty?
```

Comment:
Tried to puzzle through the logic for a few minutes, but the nested rescues turned pretty confusing. Let's leave this as-is for now.